app/react/Layout/TableRows.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+
+import { connect } from 'react-redux';
+import { TableRow } from 'app/Library/components/TableRow';
+import { IStore, TableViewColumn } from 'app/istore';
+import { IImmutable } from 'shared/types/Immutable';
+import { EntitySchema } from 'shared/types/entityType';
+import Immutable from 'immutable';
+
+interface TableRowsProps {
+  columns: TableViewColumn[];
+  storeKey: 'library' | 'uploads';
+  clickOnDocument: (e: React.SyntheticEvent, doc: EntitySchema, active: boolean) => any;
+  documents: IImmutable<{ rows: EntitySchema[] }>;
+  rowListZoomLevel: number;
+}
+
+const defaultProps = {
+  documents: Immutable.fromJS({ rows: [] }),
+  rowListZoomLevel: 2,
+};
+
+class TableRowsComponent extends Component<TableRowsProps> {
+  static defaultProps = defaultProps;
+
+  render() {
+    return (
+      <>
+        {this.props.documents.get('rows').map((entity: any) => (
+          <TableRow
+            {...{
+              entity,
+              columns: this.props.columns,
+              key: entity.get('_id'),
+              clickOnDocument: this.props.clickOnDocument,
+              storeKey: this.props.storeKey,
+              zoomLevel: this.props.rowListZoomLevel,
+            }}
```

Comment:
This is a weird use of the ellipsis syntax to use the spread operator.  Why not using the native HTML / React way of assigning keys?
```
<TableRow
   entity={entity}
   columns={this.props.columns}
   ...
/>
```

I agree this also works, and if the namings all coincide (like entity) it would work great, but since we are mostly reassigning, it doesn't make much difference.

On that note, still, doing the trick of assigning variables to props on the first line of the render method:
```
const { columns, clickOnDocument, storeKey, zoomLevel } = this.props;
```
and then just:
```
<TableRow {...{entity, clickOnDocument, storeKey, zoomLevel, key: entity.get('_id')}} />
```
could be simpler?

Just small things to repeat in every component.